utxo/src/utxo.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::Error;
+use common::{chain::TxOutput, primitives::BlockHeight};
+use serialization::{Decode, Encode};
+use std::fmt::Debug;
+
+/// Determines whether the utxo is for the blockchain of for mempool
+#[derive(Debug, Clone, Eq, PartialEq, Encode, Decode)]
+pub enum UtxoSource {
+    /// At which height this containing tx was included in the active block chain
+    BlockChain(BlockHeight),
+    MemPool,
```

Comment:
Can we rename this `MemPool` -> `Mempool`?

Reply:
Honestly, I don't like it either. And if no one objects I'll rename `BlockChain` to `Blockchain` as well

Reply:
You have my blessing too.